### PR TITLE
LPS-117816 [WIP] Add Translation component

### PR DESCRIPTION
**Pull request by dgarciasarai**
From: https://github.com/liferay-tango/liferay-portal/pull/65

### CODE REVIEW

File: modules/dxp/apps/analytics/analytics-reports-web/src/main/resources/META-INF/resources/js/components/Navigation.js [hunk 1]
```diff
@@ -47,6 +84,32 @@ export default function Navigation({
...
+	const [state, dispatch] = useReducer(dataReducer, initialState);
+
+	const safeDispatch = (action) => {
+		if (isMounted()) {
+			dispatch(action);
+		}
+	};
+
+	const handleSelectLanguageClick = (url) => {
+		safeDispatch({type: 'LOAD_DATA'});
```
Comment (dgarciasarai) [inline modules/dxp/apps/analytics/analytics-reports-web/src/main/resources/META-INF/resources/js/components/Navigation.js:98]: Loading

File: modules/dxp/apps/analytics/analytics-reports-web/src/main/resources/META-INF/resources/js/components/Navigation.js [hunk 2]
```diff
@@ -47,6 +84,32 @@ export default function Navigation({
...
+
+		fetch(url, {
```
Comment (dgarciasarai) [inline modules/dxp/apps/analytics/analytics-reports-web/src/main/resources/META-INF/resources/js/components/Navigation.js:100]: Fetch the URL of the new language

File: modules/dxp/apps/analytics/analytics-reports-web/src/main/resources/META-INF/resources/js/components/Navigation.js [hunk 3]
```diff
@@ -96,70 +159,92 @@ export default function Navigation({
...
-					<Detail
-						currentPage={currentPage}
-						languageTag={languageTag}
-						onCurrentPageChange={handleCurrentPage}
-						onTrafficSourceNameChange={handleTrafficSourceName}
-						trafficShareDataProvider={handleTrafficShare}
-						trafficVolumeDataProvider={handleTrafficVolume}
-					/>
-				)}
+					{state?.data && <>{state.data}</>}
```
Comment (dgarciasarai) [inline modules/dxp/apps/analytics/analytics-reports-web/src/main/resources/META-INF/resources/js/components/Navigation.js:245]: Load the HTML

File: modules/dxp/apps/analytics/analytics-reports-web/src/main/resources/META-INF/resources/js/AnalyticsReportsApp.js [hunk 4]
```diff
@@ -29,6 +29,7 @@ export default function ({context, props}) {
 		authorUserId,
 		publishDate,
 		title,
+		viewURLs,
```
Comment (dgarciasarai) [inline modules/dxp/apps/analytics/analytics-reports-web/src/main/resources/META-INF/resources/js/AnalyticsReportsApp.js:32]:
```js
const viewURLs = [
	{default: true, languageId: 'en-US', viewURL: 'http://...'},
	{default: false, languageId: 'pt-BR', viewURL: 'http://...'},
	{default: false, languageId: 'fr-FR', viewURL: 'http://...'},
	{default: false, languageId: 'ja-JP', viewURL: 'http://...'},
	{default: false, languageId: 'ca-ES', viewURL: 'http://...'},
	{default: false, languageId: 'es-ES', viewURL: 'http://...'},
];
```

File: modules/dxp/apps/analytics/analytics-reports-web/src/main/resources/META-INF/resources/js/components/Navigation.js [hunk 5]
```diff
@@ -97,71 +159,92 @@ export default function Navigation({
 
 	return (
 		<>
-			{!validAnalyticsConnection && (
-				<ClayAlert displayType="danger" variant="stripe">
-					{Liferay.Language.get('an-unexpected-error-occurred')}
-				</ClayAlert>
-			)}
+			{state?.loading ? (
```
Comment (wincent) [inline modules/dxp/apps/analytics/analytics-reports-web/src/main/resources/META-INF/resources/js/components/Navigation.js:162]: Why `state?.`?

`state` should always be a valid object.

**Comment by dgarciasarai** [comment]
We are going to take another approach to load the data from a language. So I'm going to close this PR and if I need review, I will send it to Frontend Infra repo. Thank you so much for your help, Greg :)

**Pull request closed without merging**